x/oracle/keeper/validatorWhiteList.go

```diff
@@ -57,20 +82,9 @@ func (k Keeper) ValidateAddress(ctx sdk.Context, address sdk.ValAddress) bool {
 	return false
 }
 
-// AddOracleWhiteList add new validator to whitelist
-func (k Keeper) AddOracleWhiteList(ctx sdk.Context, validator sdk.ValAddress) {
-	valList := k.GetOracleWhiteList(ctx)
-	k.SetOracleWhiteList(ctx, append(valList, validator))
-}
-
-// RemoveOracleWhiteList remove a validator from whitelist
-func (k Keeper) RemoveOracleWhiteList(ctx sdk.Context, validator sdk.ValAddress) {
-	valList := k.GetOracleWhiteList(ctx)
-
-	for index, item := range valList {
-		if validator.Equals(item) {
-			k.SetOracleWhiteList(ctx, append(valList[:index], valList[index+1]))
-			return
-		}
-	}
+// UpdateOracleWhiteList validator's power
+func (k Keeper) UpdateOracleWhiteList(ctx sdk.Context, networkDescriptor types.NetworkDescriptor, validator sdk.ValAddress, power uint32) {
+	valList := k.GetOracleWhiteList(ctx, networkDescriptor)
+	valList.GetWhiteList()[validator.String()] = power
```

Comment:
Let's make this line cleaner by breaking it into two lines. One line gets the whitelist into a variable, the next line sets the variable.

Reply:
accepted